esd_services_api_client/boxer/_connector.py

```diff
@@ -32,11 +31,95 @@
     ExternalAuthBase,
     RefreshableExternalTokenAuth,
 )
-from esd_services_api_client.boxer._helpers import (
-    _iterate_user_claims_response,
-    _iterate_boxer_claims_response,
-)
-from esd_services_api_client.boxer._models import BoxerClaim, UserClaim, BoxerToken
+from esd_services_api_client.boxer._helpers import _iterate_user_claims_response
+from esd_services_api_client.boxer._models import BoxerToken, Claim, ClaimPayload
+
+
+class BoxerClaimConnector:
+    """
+    Boxer Claims API connector
+    """
+
+    def __init__(self, *, base_url: str, auth: Optional[BoxerTokenAuth] = None):
+        """Creates Boxer Claims connector, capable of managing claims
+        :param base_url: Base URL for Boxer Claims endpoint
+        :param auth: Boxer-based authentication
+        """
+        self._base_url = base_url
+        self.http = session_with_retries()
+        if auth and isinstance(auth, BoxerTokenAuth):
+            self.http.hooks["response"].append(auth.get_refresh_hook(self.http))
+        self.http.auth = auth
+
+    @final
+    def get_claims(self, user_id: str, provider: str) -> Optional[Iterator[Claim]]:
+        """
+        Returns the claims assigned to the specified user_id and provider
+        """
+        response = self.http.get(f"{self._base_url}/claim/{provider}/{user_id}")
+        if response.status_code == 404:
+            return None
+        response.raise_for_status()
+        return _iterate_user_claims_response(response)
+
+    def add_user(self, user_id: str, provider: str) -> dict:
```

Comment:
on `remove_user` you have a typed response, yet here you have dict. Any reason for that?